server/src/main/java/io/deephaven/server/arrow/ArrowFlightUtil.java

```diff
@@ -326,50 +338,123 @@ public void onNext(final InputStream request) {
+                        case BarrageMessageType.BarrageSubscriptionRequest:
+                            serviceBarrageSubscriptionRequest(message);
+                            break;
+                        case BarrageMessageType.DoGetRequest: // rename to SnapshotRequest?
+                            serviceDoGetRequest(message);
+                            break;
+                        default:
+                            log.warn().append(myPrefix).append("received a message with unhandled BarrageMessageType")
+                                    .endl();
+                            return;
                     }
-                    final BarrageSubscriptionRequest subscriptionRequest = BarrageSubscriptionRequest
-                            .getRootAsBarrageSubscriptionRequest(message.app_metadata.msgPayloadAsByteBuffer());
+                }
+            });
+        }
 
-                    if (bmp != null) {
-                        apply(subscriptionRequest);
-                        return;
-                    }
+        private void serviceDoGetRequest(BarrageProtoUtil.MessageInfo message) {
+            final DoGetRequest doGetRequest = DoGetRequest
+                    .getRootAsDoGetRequest(message.app_metadata.msgPayloadAsByteBuffer());
+
+            final SessionState.ExportObject<BaseTable> parent =
+                    ticketRouter.resolve(session, doGetRequest.ticketAsByteBuffer(), "ticket");
+
+            final BarrageSubscriptionRequest subscriptionRequest = BarrageSubscriptionRequest
+                    .getRootAsBarrageSubscriptionRequest(message.app_metadata.msgPayloadAsByteBuffer());
+
+            session.nonExport()
+                    .require(parent)
+                    .onError(listener)
+                    .submit(() -> {
+                        final BaseTable table = parent.get();
+
+                        // Send Schema wrapped in Message
+                        final FlatBufferBuilder builder = new FlatBufferBuilder();
+                        final int schemaOffset = BarrageUtil.makeSchemaPayload(builder, table.getDefinition(),
+                                table.getAttributes());
+                        builder.finish(MessageHelper.wrapInMessage(builder, schemaOffset,
+                                org.apache.arrow.flatbuf.MessageHeader.Schema));
+                        final ByteBuffer serializedMessage = builder.dataBuffer();
+
+                        // leverage the stream generator SchemaView constructor
+                        final BarrageStreamGenerator.SchemaView schemaView =
+                                new BarrageStreamGenerator.SchemaView(serializedMessage);
+
+                        // push the schema to the listener
+                        listener.onNext(schemaView);
+
+                        // collect the viewport and columnsets (if provided)
+                        final boolean hasColumns = subscriptionRequest.columnsVector() != null;
```

Comment:
aside from the columns and rowset, can we share more of the implementation with customDoGet? That is, let customDoGet call into some shared code, with no columns/rows set?

this probably makes sense to do after splitting the subscribe/doget.